### ISSUE TITLE
Knex types not being inferred

### ISSUE THREAD

**Issue by evert** [knex/knex#5156]
# Environment

Knex version: 3
Database + version: MSQL
OS: Linux

@lorefnon.

# Bug

I have 2 projects, both using knex and typescript. Both are open source. One of the two is simply not listening to my Knex table types. My definitions look like this:

```typescript
declare module 'knex/types/tables' {
  interface Principal {
    id: number;
    identity: string;
    type: number;
    external_id: string;
    nickname: string;
    created_at: number;
    modified_at: number;
    active: number;
  }

  interface OAuth2Client {
    id: number;
    client_id: string;
    client_secret: string;
    allowed_grant_types: string;
    user_id: number;
    require_pkce: number;
  }

  interface Tables {
    oauth2_clients: OAuth2Client;
    principals: Principal;
  }
}
```

I've placed this file in my `knexfile.ts`, in the file that sets up the `knex` global singleton, and in a separate knex-definitions.

Elsewhere in my code, I have code like this:

```typescript
  const result = await db('oauth2_clients')
    .select('*')
    .where('client_id', clientId);
```

So far `result` is not inferred as `OAuth2Client[]`, and I'm not sure what else to share to get close to the root of this.

**Comment by lorefnon** [comment]
Before `declare module ...`, can you add an `import 'knex'` at the top and see if it resolves the issue ?

**Comment by lorefnon** [comment]
Sorry you need to import `knex/types/tables` - TS is a bit finicky wrt module augmentation.

Interesting project btw. Enterprise auth & identity management is something I have been planning to spend time digging into this year.

**Comment by evert** [comment]
Also no luck, lol. I also tried making sure that I'm consistently importing knex as `import knex from 'knex'` and remove instances where I did `import { knex } from 'knex'`, because I was worried that _somehow_ Typescript would consider the 2 import paths as separate.

Maybe I just need to look at my [other project](https://github.com/badgateway/tt-api/blob/main/src/knex.ts) (where this does work!) and reverse engineer it until I find what's the difference between the two. 

> Interesting project btw. Enterprise auth & identity management is something I have been planning to spend time digging more into this year.

Thanks! Well let me know if you ever want a chat about it! I like meeting new people!
